### ISSUE TITLE
Add HTTP Basic Auth support

### ISSUE THREAD

**Issue by klein0r** [ioBroker/ioBroker.simple-api#45]
Please add support for HTTP Basic authentication.
Just extract the user information from the auth header for validation.

**Comment by dadude86** [comment]
would be awesome

**Comment by mesche** [comment]
Authentication password via url parameter is not secure

**Comment by Apollon77** [comment]
@mesche hm ... with SSL it could be ... but still think about most do it home-network ...

**Comment by mesche** [comment]
Correct... but when I use the API with other systems (like Grafana for example), then I have to store the password in plain text.
--> not quite ideal  ;-)

but thank your for your support and the development .. great work

**Comment by SirFlour** [comment]
@mesche How do you connect grafana with iobroker when auth is activated? i am trying with the grafana json plugin and iobroker simple api. it is working without auth activated, but when i activate it shows bad request. i can get value directly via url, so simple api is working. can you share your settings? many thanks and have a nice day

**Comment by mesche** [comment]
> @mesche How do you connect grafana with iobroker when auth is activated? i am trying with the grafana json plugin and iobroker simple api. it is working without auth activated, but when i activate it shows bad request. i can get value directly via url, so simple api is working. can you share your settings? many thanks and have a nice day

Not at all! That's why I hope that the simple-api adapter supports this function soon.

Currently I have 2 instances of the simple-api adapter running. For Grafana without auth and only with read permissions of the data points

**Comment by SirFlour** [comment]
Ahhh now i understand you. Allright, your workaround with 2 instances is great. I will do one just for read and only available between the dockers containers. and the other one with auth and extra port. 

i hope it can be added so that this workaround is not needed any more and simple-api gets better :)

**Comment by tommyMX** [comment]
Any progress here ?

**Comment by SimonFischer04** [comment]
> Any progress here ?

https://github.com/ioBroker/ioBroker.rest-api already supports this. Wouldn't it be better to focus on one adapter instead of duplicating all the work in two adapters that ultimately provide the same basic functionality.

Anything simple-api can do more than rest-api? Or could simple-api just be deprecated and suggested to move to rest-api?

**Comment by Apollon77** [comment]
If you can migrate to rest-api because thats a proper REST-API and not a bit mixed stuff like simple-api :-))
